### DISCUSSION TITLE
Diagnostics

### DISCUSSION THREAD

**Discussion by nordlow** [ccache/ccache#138] [Support]
We have a problem with incorrect results in one of our ccache-enable builds. The situation is quite complex and seems to depend on invalid cache lookups of the dependency file generated by GCC.

In order to more easily understand what goes wrong we would like to add some kind of optional diagnostics to ccache. Typically printing which files are re-compiled, copied to cache, fetched from cache and what hash-keys that are involved.

Any suggestions on how to add this?

**Comment by afbjorklund** [comment]
The log file should have all that (and then some).

**Comment by nordlow** [comment]
I just noticed that. Thanks!
